cluster-autoscaler/cloudprovider/aws/README.md

```diff
@@ -26,29 +26,37 @@ A minimum IAM policy would look like:
 }
 ```
 
-If you'd like to auto-discover node groups by specifing the `--node-group-auto-discover` flag, a `DescribeTags` permission is also required:
+If you'd like to auto-discover node groups by specifing the `--node-group-auto-discovery` flag, a `DescribeTags` permission is also required. We highly recommend you specify the ASGs explicitly in your IAM Roles for production systems. The autoscaler will only be able to modify ASGs you give it access to.
 
 ```json
 {
     "Version": "2012-10-17",
     "Statement": [
+        {
+            "Effect": "Allow",
+            "Action": [
+                "autoscaling:SetDesiredCapacity",
+                "autoscaling:TerminateInstanceInAutoScalingGroup"
+            ],
+            "Resource": [
+                "arn:aws:autoscaling:<REGION>:<ACCOUNTID>:autoScalingGroup:*:autoScalingGroupName/<ASG-1-NAME>",
+                "arn:aws:autoscaling:<REGION>:<ACCOUNTID>:autoScalingGroup:*:autoScalingGroupName/<ASG-2-NAME>"
```

Comment:
I'd rather like to set `Condition` instead, so that we don't need to repeat ASG names in the IAM policy - my original intention to introduce auto-discovery was to reduce repetitions!

```yaml
                 "Condition": {
                    "Null": { "autoscaling:ResourceTag/kubernetes.io/cluster/{{.ClusterName}}": "false" }
                  }
```

Note: replace `{{.ClusterName}}` with your cluster name.

Actually, this is what kube-aws has been using since disasters of CA+Auto-Discovery terminating nodes unrelated to the k8s cluster.